packages/data-context/src/sources/VersionsDataSource.ts

```diff
@@ -51,20 +57,24 @@ export class VersionsDataSource {
    * }
    */
   async versionData (): Promise<VersionData> {
-    const [latestVersion, npmMetadata] = await Promise.all([this._latestVersion, this._npmMetadata])
+    const versionData = this.#ensureData()
+    const [latestVersion = pkg.version, npmMetadata] = await Promise.all([
```

Comment:
Why do we need to default this to pkg.version. Doesn't `latestVersion` always return a string

Reply:
I think this was a type issue, but I will update this

Reply:
Ah I guess since it's getting it from ensureData as `this.ctx.coreData.versionData` there may not be a guarantee they're populated.

Reply:
Yeah, to make it correctly typesafe without cheating, the alternative was uglier I think:

```ts
  #ensureData () {
    this.ctx.update((d) => {
      d.versionData.latestVersion ??= this.getLatestVersion().catch((e) => pkg.version)
      d.versionData.npmMetadata ??= this.getVersionMetadata().catch((e) => undefined)
    })

    return this.ctx.coreData.versionData
  }
```
becomes

```ts
  #ensureData () {
    this.ctx.update((d) => {
      d.versionData.latestVersion ??= this.getLatestVersion().catch((e) => pkg.version)
      d.versionData.npmMetadata ??= this.getVersionMetadata().catch((e) => undefined)
    })

    const { latestVersion, npmMetadata } = this.ctx.coreData.versionData

    assert(latestVersion)

    return { latestVersion, npmMetadata }
  }
```

Up to you if you think we should change. We could also look at a different pattern of doing this, similar to `GitDataSource` where we pass in what we need and cache that in the `versionData`, making it possible to reset without many changes here

Reply:
Yeah that's a little messier. I think I'm ok with the way you have it and it makes sense. It just stood out to me a little bit was the main thing